client/src/components/graph/overlays/centroidLabels.js

```diff
@@ -2,42 +2,114 @@ import React, { PureComponent } from "react";
 import { connect } from "react-redux";
 
 import { categoryLabelDisplayStringLongLength } from "../../../globals";
+import calcCentroid from "../../../util/centroid";
+import { createColorQuery } from "../../../util/stateManager/colorHelpers";
 
 export default
 @connect((state) => ({
-  colorAccessor: state.colors.colorAccessor,
+  annoMatrix: state.annoMatrix,
+  colors: state.colors,
+  layoutChoice: state.layoutChoice,
   dilatedValue: state.pointDilation.categoryField,
-  labels: state.centroidLabels.labels,
   categoricalSelection: state.categoricalSelection,
+  showLabels: state.centroidLabels?.showLabels,
 }))
 class CentroidLabels extends PureComponent {
-  // Check to see if centroids have either just been displayed or removed from the overlay
+  constructor(props) {
+    super(props);
+    this.state = {
+      status: "pending",
+      labels: new Map(),
+      colorAccessor: null,
+    };
+  }
+
+  componentDidMount() {
+    this.updateState(null);
+  }
 
   componentDidUpdate(prevProps) {
```

Comment:
I did not know about that new lifecycle - interesting.  Certainly easier to think about.

In this case I'm not sure it will have the expected benefit (of removing rendering) because the updateState() is async (it calls fetch).  That said, it is a cleaner pattern.

Is this a PR blocker, or something to do in the next pass.  I'm fine with either, but have a slight preference for moving stylistic stuff to a future PR (we can capture this in an issue so as to not lose it).  Please comment.

Reply:
Not a blocker, just wanted to bring to attention 👍

Reply:
Filed as #1624 - will catch in a subsequent PR.

Reply:
Also a relevant article on `getDerivedStateFromProps()`: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html just in case!

Reply:
added to #1624